src/main.js

```diff
@@ -10,20 +10,99 @@ Config.set({
         timeout: parseInt(3000), // 90000 or less
     },
     realmPath: 'alpha', // e.g. 'alpha' or 'root'
-    // tree: process.env.TREE, // e.g. 'sdkAuthenticationTree' or 'Login'
+    tree: 'AJ Choice Login', // e.g. 'sdkAuthenticationTree' or 'Login'
 });
 
+const loginElem = document.getElementById('login-form');
+const continueElem = document.getElementById('continue-form');
+const successElem = document.getElementById('success');
+const errorElem = document.getElementById('error');
+const userPanel = document.getElementById('user');
+const userInfoElem = document.querySelector('#user pre');
+
+async function nextSteps(step) {
+    do {
+        const nextStep = await FRAuth.next(step);
+        console.log('nextStep', nextStep);
+        await handleNextStep(nextStep);
+        step = nextStep;
+    } while (step);
+    return;
+}
+
+function getStage(step) {
+    let stage;
+    const nameCallbacks = step.getCallbacksOfType('NameCallback');
+    const passwordCallbacks = step.getCallbacksOfType('PasswordCallback');
+    const choiceCallbacks = step.getCallbacksOfType('ChoiceCallback');
+    
+    if (nameCallbacks.length && passwordCallbacks.length) {
+        stage = 'UsernamePassword';
+    } else if (choiceCallbacks.length) {
+        stage = 'Choice';
+    } else {
+        stage = undefined;
+    }
+    
+    console.log('stage', stage);
+    return stage;
+}
+
+function renderChoices(step) {
+    const choiceCallback = step.getCallbackOfType('ChoiceCallback');
+    // console.log('choiceCallback', choiceCallback);
+    const prompt = choiceCallback.getPrompt();
+    const choices = choiceCallback.getChoices();
+    const defaultChoice = choiceCallback.getDefaultChoice();
+    // console.log('choice', prompt, choices);
+
+    if (!prompt || !choices.length) {
+        throw new Error('No prompt or choices found');
+    }
+
+    const promptElem = document.createElement('p');
+    promptElem.innerHTML = prompt;
+    continueElem.appendChild(promptElem);
+
+    choices.forEach(choice => {
+        const radio = document.createElement('div');
+        const input = document.createElement('input');
+        input.setAttribute('type', 'radio');
+        input.setAttribute('id', choice);
+        input.setAttribute('name', 'choice');
+        input.setAttribute('value', choice);
+        input.setAttribute('checked', defaultChoice === choice);
+        radio.appendChild(input);
+
+        const label = document.createElement('label');
+        label.setAttribute('for', choice);
+        label.innerHTML = choice;
+        radio.appendChild(label);
+
+        continueElem.appendChild(radio);
+        continueElem.addEventListener('submit', () => submitChoiceHandler(step));
```

Comment:
One way to simplify this is to use only one form element itself, and just swap out its contents. That way, you have a single form submission event handler, and it acts as your iterator. Within the submission callback function, you would then call `FRAuth.next` and handle the response, deciding if it has a step, success or failure, then calling the appropriate handler function.